en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,126 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some blocks will require environment parameters.
+Those parameters could be of multiple types, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when the block is called.
+
+This easy solution is not optimal for a main reason: when loading blocks via `ESI`_, each
+block will be cached for each url variant.
```

Comment:
"easy way" and "easy solution" sound halfway between convoluted and condescending. I do not like it at all.